nexus/db-queries/src/db/datastore/switch_port.rs

```diff
@@ -312,319 +316,2021 @@ impl DataStore {
+                        error!(opctx.log, "{message}"; "error" => ?external_error);
+                        external_error
+                    }
+                    None => {
+                        error!(opctx.log, "{message}"; "error" => ?e);
+                        Error::internal_error(
+                            "error while removing entry from allow export list",
+                        )
+                    }
+                }
+            })
+    }
 
-                result.interfaces =
-                    interface_config_dsl::switch_port_settings_interface_config
-                        .filter(interface_config::port_settings_id.eq(id))
-                        .select(SwitchInterfaceConfig::as_select())
-                        .load_async::<SwitchInterfaceConfig>(&conn)
-                        .await?;
+    pub async fn switch_port_configuration_bgp_peer_community_list(
+        &self,
+        opctx: &OpContext,
+        configuration: NameOrId,
+        bgp_peer: IpAddr,
+    ) -> ListResultVec<SwitchPortBgpPeerConfigCommunity> {
+        use db::schema::switch_port_settings as port_settings;
+        use db::schema::switch_port_settings_bgp_peer_config_communities as communities;
+
+        let dataset = port_settings::table.inner_join(
+            communities::table
+                .on(communities::port_settings_id.eq(port_settings::id)),
+        );
 
-                use db::schema::switch_vlan_interface_config as vlan_config;
-                use db::schema::switch_vlan_interface_config::dsl as vlan_dsl;
-                let interface_ids: Vec<Uuid> = result
-                    .interfaces
-                    .iter()
-                    .map(|interface| interface.id)
-                    .collect();
+        let query = match configuration {
+            NameOrId::Id(id) => {
+                // find port config using port settings id
+                dataset.filter(port_settings::id.eq(id)).into_boxed()
+            }
+            NameOrId::Name(name) => {
+                // find port config using port settings name
+                dataset
+                    .filter(port_settings::name.eq(name.to_string()))
+                    .into_boxed()
+            }
+        };
 
-                result.vlan_interfaces = vlan_dsl::switch_vlan_interface_config
-                    .filter(vlan_config::interface_config_id.eq_any(interface_ids))
-                    .select(SwitchVlanInterfaceConfig::as_select())
-                    .load_async::<SwitchVlanInterfaceConfig>(&conn)
-                    .await?;
+        query
+            .filter(communities::addr.eq(IpNetwork::from(bgp_peer)))
+            .select(SwitchPortBgpPeerConfigCommunity::as_select())
+            .load_async(&*self.pool_connection_authorized(opctx).await?)
+            .await
+            .map_err(|e: diesel::result::Error| {
+                let msg = "error while looking up bgp peer allowed export list";
```

Comment:
This looks like a copy/paste issue, since I don't expect export here, but expect community.

Check other places just in case.